File: WEEK22/DFS_네트워크.py
def solution(n, computers):
    answer = 0
    visited = [False for _ in range(n)]
    
    def depth_first(graph, node, visited):
        visited[node] = True;
        
        for index, value in enumerate(graph[node]):
            if index != node:
                if (visited[index] == False) & (value == 1):
                    depth_first(graph, index, visited)
    
    for index, value in enumerate(visited):
        if value == False:
            answer += 1
            
            depth_first(computers, index, visited)
    
    return answer

File: WEEK22/DFS_네트워크.js
function solution(n, computers) {
  var answer = 0;
  let visited = new Array(n).fill(false);

  function depthFirst(graph, node, visited) {
    visited[node] = true;

    graph[node].forEach((element, index) => {
      if (index !== node) {
        if ((visited[index] === false) & (element === 1)) {
          depthFirst(graph, index, visited);
        }
      }
    });
  }

  visited.forEach((element, index) => {
    if (element === false) {
      answer++;

      depthFirst(computers, index, visited);
    }
  });

  return answer;
}

console.log(
  solution(3, [
    [1, 1, 0],
    [1, 1, 0],
    [0, 0, 1],
  ])
);
